spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -53,6 +58,7 @@ abstract class BaseBatchReader<T extends ScanTask> extends BaseReader<ColumnarBa
 
   protected CloseableIterable<ColumnarBatch> newBatchIterable(
       InputFile inputFile,
+      ByteBuffer keyMetadata,
```

Comment:
Looks like these changes can't be made to the corresponding `BaseBatchReader` in `iceberg-arrow` because it is public. I think that means that Arrow cannot support reading from encrypted tables in this PR even if it calls the `EncryptionManager` correctly.